rq/job.py

```diff
@@ -565,7 +564,12 @@ def save(self, pipeline=None, include_meta=True):
         key = self.key
         connection = pipeline if pipeline is not None else self.connection
 
-        hmset(connection, key, self.to_dict(include_meta=include_meta))
+        mapping = self.to_dict(include_meta=include_meta)
+
+        if redis_min_version(self.connection, "4.0.0"):
```

Comment:
`redis_min_version` incurs an extra `redis.info` call everytime `job.save()` is called, which is multiple times in every single job lifecycle.

To avoid this, worker should assign `job.redis_server_version` when job is dequeued. `job.save()` should look something like this.

```python
if not self.redis_server_version:
    self.redis_server_version = connection.info("server")["redis_version"]

if self.redis_server_version >= '4.0.0':  # Do a proper version check
   connection.hset(key, mapping=mapping)
else:
    connection.hmset(key, mapping)
```

Reply:
Please check the latest changes, it applies what you suggest.